share/functions/prompt_pwd.fish

```diff
@@ -24,11 +24,24 @@
     set -q fish_prompt_pwd_full_dirs
     or set -l fish_prompt_pwd_full_dirs 1
 
+    set -ql _flag_p
+    and set -l fish_prompt_pwd_dir_depth $_flag_p
+
+    set -q fish_prompt_pwd_dir_depth
+    or set -l fish_prompt_pwd_dir_depth 0
+
     for path in $argv
         # Replace $HOME with "~"
         set -l realhome (string escape --style=regex -- ~)
         set -l tmp (string replace -r '^'"$realhome"'($|/)' '~$1' $path)
 
+        # Shorten to at most $fish_prompt_pwd_dir_depth components, unless set to 0.
+        if test "$fish_prompt_pwd_dir_depth" -gt 0
+            set -l split (string split / $tmp)
+            set -l length (math (count $split) - (math $fish_prompt_pwd_dir_depth - 1))
+            set tmp $split[$length..]
```

Comment:
This misbehaves if length is 0 or negative.
We should probably fix this by doing
something to the effect of `set length (math "min($length, 1)")`

	$ prompt_pwd --dir-depth=1
	fish-shell
	$ prompt_pwd --dir-depth=2
	git/fish-shell
	$ prompt_pwd --dir-depth=3
	~/git/fish-shell
	$ prompt_pwd --dir-depth=4
	embedded:functions/prompt_pwd.fish (line 42): array indices start at 1, not 0.
	            set tmp $split[$length..]
	                            ^
	in function 'prompt_pwd' with arguments '--dir-depth=4'
	~/g/fish-shell
	$ prompt_pwd --dir-depth=5
	fish-shell
	[1] $ prompt_pwd --dir-depth=6
	git/fish-shell

(also note that `--dir-depth=5` results in a spurious exit code of 1, which we shoudl fix)